File: server/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/IssueReportTypesUpdateRequest.java
package edu.cnm.deepdive.seesomethingabq.model.dto;

import java.util.List;

/**
 * Request body DTO for replacing the issue types associated with an issue report.
 */
public class IssueReportTypesUpdateRequest {

  private List<String> issueTypeTags;

  /**
   * Returns the replacement set of issue type tags.
   *
   * @return issue type tags.
   */
  public List<String> getIssueTypeTags() {
    return issueTypeTags;
  }

  /**
   * Sets the replacement set of issue type tags.
   *
   * @param issueTypeTags issue type tags.
   */
  public void setIssueTypeTags(List<String> issueTypeTags) {
    this.issueTypeTags = issueTypeTags;
  }

}

File: app/src/main/java/edu/cnm/deepdive/seesomethingabq/model/dto/IssueReportTypesUpdateRequest.kt
package edu.cnm.deepdive.seesomethingabq.model.dto

import com.google.gson.annotations.SerializedName

/**
 * Request DTO for replacing the issue-type tags of an issue report (manager-only).
 *
 * @property issueTypeTags replacement list of issue-type tags.
 */
data class IssueReportTypesUpdateRequest(
  @SerializedName("issueTypeTags")
  val issueTypeTags: List<String>
)

